common/src/main/java/org/mcaccess/minecraftaccess/features/inventory_controls/GroupGenerator.java

```diff
@@ -358,32 +359,35 @@ public static List<SlotsGroup> generateGroupsFromSlots(AbstractContainerScreenAc
         //<editor-fold desc="Group enchantment screen enchant buttons (EnchantScreen.java -->> render())">
         if (Minecraft.getInstance().player != null && Minecraft.getInstance().level != null
                 && screen.getMenu() instanceof EnchantmentMenu enchantmentScreenHandler) {
-            int i = enchantmentScreenHandler.getGoldCount();
-            for (int j = 0; j < 3; ++j) {
-                int k = enchantmentScreenHandler.costs[j];
-                int l = enchantmentScreenHandler.levelClue[j];
-                // copied from 1.21.3 EnchantmentScreen.render() L172
-                Optional<Holder.Reference<Enchantment>> optional = Minecraft.getInstance()
-                        .level
-                        .registryAccess()
-                        .lookupOrThrow(Registries.ENCHANTMENT)
-                        .get(l);
-
-                int m = j + 1;
-                if (optional.isEmpty()) break;
-                StringBuilder clueText = new StringBuilder(Component.translatable("container.enchant.clue", Enchantment.getFullname(optional.get(), l)).withStyle(ChatFormatting.WHITE).getString());
+            for (int i = 0; i < 3; i++) {
+                int enchantmentId = enchantmentScreenHandler.enchantClue[i];
+                int requiredLevel = enchantmentScreenHandler.costs[i];
+                int level = enchantmentScreenHandler.levelClue[i];
+                int cost = i + 1;
+
+                Optional<Holder.Reference<Enchantment>> enchantment = Minecraft.getInstance().level.registryAccess().lookupOrThrow(Registries.ENCHANTMENT).get(enchantmentId);
+                if (enchantment.isEmpty()) break;
+
+                StringBuilder clueText = new StringBuilder(Component.translatable("container.enchant.clue", Enchantment.getFullname(enchantment.get(), level)).getString()); // this breaks when using I18n.get() for some reason so Component.translatable() is being used instead
```

Comment:
The reason that stuff breaks is that `Enchantment.getFullname()` returns a `Component`. This can be fixed by adding `.getString()` to it but that's only marginally better than whats already happening so it doesn't really matter too much, this will be refactored as part of #369 anyway.
```suggestion
                StringBuilder clueText = new StringBuilder(I18n.get("container.enchant.clue", Enchantment.getFullname(enchantment.get(), level).getString()));
```